sub/osd.c

```diff
@@ -515,10 +515,14 @@ void osd_rescale_bitmaps(struct sub_bitmaps *imgs, int frame_w, int frame_h,
     int cy = vidh / 2 - (int)(frame_h * yscale) / 2;
     for (int i = 0; i < imgs->num_parts; i++) {
         struct sub_bitmap *bi = &imgs->parts[i];
-        bi->x = (int)(bi->x * xscale) + cx + res.ml;
-        bi->y = (int)(bi->y * yscale) + cy + res.mt;
-        bi->dw = (int)(bi->w * xscale + 0.5);
-        bi->dh = (int)(bi->h * yscale + 0.5);
+        int x1 = lrint(bi->x * xscale);
+        int y1 = lrint(bi->y * yscale);
+        int x2 = lrint((bi->x + bi->w) * xscale);
+        int y2 = lrint((bi->y + bi->h) * yscale);
+        bi->x = x1 + cx + res.ml;
+        bi->y = y1 + cy + res.mt;
+        bi->dw = x2 - x1;
+        bi->dh = y2 - y1;
```

Comment:
nit, but maybe it is more readable to use `mp_rect`?
```suggestion
        struct mp_rect rc = {
            .x0 = lrint(bi->x * xscale),
            .y0 = lrint(bi->y * yscale),
            .x1 = lrint((bi->x + bi->w) * xscale),
            .y1 = lrint((bi->y + bi->h) * yscale),
        };
        bi->x = rc.x0 + cx + res.ml;
        bi->y = rc.y0 + cy + res.mt;
        bi->dw = mp_rect_w(rc);
        bi->dh = mp_rect_h(rc);
```